guides/advanced_guides/keys.md

```diff
@@ -1,145 +1,38 @@
 # Keys
 
-To communicate with MeiliSearch's `RESTful API` most of the routes require an `API key`.
+In MeiliSearch there are three types of keys:
 
-The following information defines a key:
-* A **[Key](/guides/advanced_guides/keys.md#key)** generated by MeiliSearch
-* A **[Description](/guides/advanced_guides/keys.md#description)**
-* An **[ACL:](/guides/advanced_guides/keys.md#acl)** list of permissions
-* An **[Indexes list](/guides/advanced_guides/keys.md#indexes)**  with wildcard on which the ACL's are applied
-* A **[Revoked](/guides/advanced_guides/keys.md#revoked)** updatable boolean to revoke a key <Badge text="soon" type="warn"/>
-* An **[Expires_at](/guides/advanced_guides/keys.md#expires-at)** timestamp that indicated the expire date of the token
+- **Master** key: access to all routes.
+- **Private** key: access to all routes except the `/keys` routes.
+- **Public** key: only access:
+    - `GET /indexes/search`
+    - `GET /indexes/:index_uid/search`
+    - `GET /indexes/:index_uid/documents`
+    - `GET /indexes/:index_uid/documents/:doc_id`
 
-A key is passed to MeiliSearch using the header
-` X-Meili-Api-Key: myApiKey `
+When a master key is given, MeiliSearch will have one master key and generate one private key and one public key. **You can not create additionals keys**.
 
-## Master key
+### Master Key
 
-This is the only key that is not created by the API but is **set as environement variable or as a binary flag** when [launching the MeiliSearch binary](/guides/advanced_guides/binary.md).
+When launching an instance of MeiliSearch, you have the possibility to give a master key. By providing a master key, all routes in MeiliSearch will be protected and will require a key to access.
 
-This key gives access to every route of the API.
+You can communicate it as:
+- an environment variable: `MEILI_MASTER_KEY=xxx`
+- a CLI option: `./meilisearch --master-key "xxx"`
 
-::: warning
-When no master key is set on binary launch, no API key is needed on any route.
-**Giving full access to the API**.
-:::
+With this master key, you can retrieve the private key and the public using the [keys route](/references/keys.md).
 
-#### Examples
+On each API call, the key must be added in [the header](/references/#authentication). Depending on the provided key you will have different permissions ([see above](/guides/advanced_guides/keys.md)).
 
------
+#### No master key
 
-```bash
-# with the flag
-$ ./meilisearch --api-key myMasterKey
-# with the environement variable
-$ MEILI_API_KEY=myMasterKey ./meilisearch
-```
+If no master key is given, all routes in MeiliSearch can be accessed without a key.
 
-This header `X-Meili-Api-Key: myMasterKey` gives access to the whole API.
+### Reset keys
 
------
-```bash
-$ ./meilisearch
-```
+Since the private and public keys are generated based on your master key, you must change the master key to reset these both keys.<br>
```

Comment:
```suggestion
Since the private, and public keys are generated based on your master key, you must change the master key to reset the two other ones.

```